x-pack/plugins/maps/public/connected_components/edit_layer_panel/join_editor/resources/spatial_join_expression/spatial_join_popover_content.tsx

```
@@ -0,0 +1,142 @@
+import type {
+  ESDistanceSourceDescriptor,
+  JoinSourceDescriptor,
+} from '../../../../../../common/descriptor_types';
+import { getIndexPatternService } from '../../../../../kibana_services';
+import { getGeoFields } from '../../../../../index_pattern_util';
+import { GeoIndexPatternSelect } from '../../../../../components/geo_index_pattern_select';
+import { GeoFieldSelect } from '../../../../../components/geo_field_select';
+import { inputStrings } from '../../../../input_strings';
+import { RelationshipExpression } from '../../../../../classes/layers/wizards/spatial_join_wizard';
+import { DEFAULT_WITHIN_DISTANCE } from '../../../../../classes/sources/join_sources';
+
+interface Props {
+  sourceDescriptor: Partial<ESDistanceSourceDescriptor>;
+  onSourceDescriptorChange: (sourceDescriptor: Partial<JoinSourceDescriptor>) => void;
+}
+
+export function SpatialJoinPopoverContent(props: Props) {
+  const [rightDataView, setRightDataView] = useState<DataView | undefined>(undefined);
+  const [rightGeoFields, setRightGeoFields] = useState<DataViewField[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [unableToLoadDataView, setUnableToLoadDataView] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (props.sourceDescriptor.indexPatternId === undefined) {
+      return;
+    }
+
+    let ignore = false;
+    setIsLoading(true);
+    getIndexPatternService()
+      .get(props.sourceDescriptor.indexPatternId)
+      .then((dataView) => {
+        if (!ignore) {
+          setIsLoading(false);
+          setRightDataView(dataView);
+          setRightGeoFields(getGeoFields(dataView.fields));
+        }
+      })
+      .catch((err) => {
+        if (!ignore) {
+          setIsLoading(false);
+          setUnableToLoadDataView(true);
+        }
+      });
+
+    return () => {
+      ignore = true;
+    };
+    // only run on mount
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Disabling this eslint rule seems like a code smell, but I feel like I know enough about react hooks to confirm. 🤷

---

Agreed that the linting rule is a little too strict. Do we have any consensus on whether using `useMount` is more appropriate in these simple situations?

---

Just sliding in here with a note since I've hit this in the past: `useMount` doesn't support a return callback like it is used with `useEffect` here for being called on unmount so it's unsuitable unfortunately (and it's just silently dropping it if a return callback is present so easy to miss).